File: Models/TutorialStep.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace ByodLauncher.Models
{
    public class TutorialStep
    {
        public Guid Id { get; set; }

        public int SequenceNumber { get; set; }

        public string Title { get; set; }

        public string Instruction { get; set; }

        public Guid TutorialTargetId { get; set; }
        public TutorialTarget TutorialTarget { get; set; }
    }
}

File: wwwroot/src/models/tutorialStep.ts
import {Id} from "@/models/idType";

export class TutorialStep {
    id: Id = null;
    sequenceNumber = 0;
    title = '';
    instruction = '';
    tutorialTargetId: Id = null;

    public constructor(init?: Partial<TutorialStep>) {
        Object.assign(this, init);
    }

    public update(updateData: Partial<TutorialStep>) {
        Object.assign(this, updateData);
    }
}